### PR TITLE
Impact Analysis - CMS177, CMS645, CMS646, CMS771, CMS951

### PR DESCRIPTION
**Pull request by soy-park**
Impact analysis of QICore6 uplift for:
- **CMS177** (_Child and Adolescent Major Depressive Disorder (MDD): Suicide Risk Assessment_)
   - No significant changes noted 
- **CMS645** (_Bone density evaluation for patients with prostate cancer and receiving androgen deprivation therapy_)
   - Added comments to update to `LaboratoryResultObservation`, `ObservationCancelled`, `ConditionProblemsHealthConcerns`, and `ConditionEncounterDiagnosis`
- **CMS646** (_Intravesical Bacillus-Calmette-Guerin for non-muscle invasive bladder cancer_)
   - Added comments to update to `ConditionEncounterDiagnosis`, `ConditionProblemsHealthConcerns`, and `LaboratoryResultObservation`
- **CMS771** (_Urinary Symptom Score Change 6-12 Months After Diagnosis of Benign Prostatic Hyperplasia_)
   - Added comments to update to `USCoreBMIProfile`, `ObservationScreeningAssessment`, `ConditionProblemsHealthConcerns`, and `ConditionEncounterDiagnosis`
- **CMS951** (_Kidney Health Evaluation_)
   - Added comments to update to `LaboratoryResultObservation`, `ConditionProblemsHealthConcerns`, and `ConditionEncounterDiagnosis`

### CODE REVIEW

File: input/cql/BoneDensityProstateCancerAndrogenDeprivationTherapyFHIR.cql [hunk 1]
```diff
@@ -142,6 +153,9 @@ define "Has Qualifying Encounter":
         and OfficeVisit.status = 'finished'
   )
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/BoneDensityProstateCancerAndrogenDeprivationTherapyFHIR.cql:158]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/Hospice.cql [hunk 2]
```diff
@@ -54,9 +51,8 @@ define "Has Hospice Services":
     )
 
     /*
-     @QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+    @QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
```
Comment (jbyrdevans) [inline input/cql/Hospice.cql:54]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql [hunk 3]
```diff
@@ -80,6 +84,9 @@ define "July 1 of Year Prior to the Measurement Period":
 define "June 30 of the Measurement Period":
   DateTime((year from start of "Measurement Period"), 6, 30, 23, 59, 59, 0, 0)
 
+/*
+@QICore6Uplift: Update Condition to ConditionEncounterDiagnosis
+*/
```
Comment (jbyrdevans) [inline input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql:89]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql [hunk 4]
```diff
@@ -92,6 +99,9 @@ define "Cystectomy Done":
       such that Cystectomy.performed.toInterval ( ) 6 months or less before start of FirstBladderCancerStaging.performed.toInterval ( )
     where Cystectomy.status = 'completed'
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql:104]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql [hunk 5]
```diff
@@ -135,6 +145,9 @@ define "Has Qualifying Encounter":
         and ValidEncounter.status = 'finished'
   )
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql:150]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/KidneyHealthEvaluationFHIR.cql [hunk 6]
```diff
@@ -79,12 +86,18 @@ define "Has Outpatient Visit During Measurement Period":
         and ValidEncounter.status = 'finished'
   )
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/KidneyHealthEvaluationFHIR.cql:91]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/KidneyHealthEvaluationFHIR.cql [hunk 7]
```diff
@@ -79,12 +86,18 @@ define "Has Outpatient Visit During Measurement Period":
...
+*/ 
 define "Has Active Diabetes Overlaps Measurement Period":
   exists ( ["Condition": "Diabetes"] Diabetes
       where Diabetes.prevalenceInterval ( ) overlaps day of "Measurement Period"
         and Diabetes.clinicalStatus ~ "active"
   )
 
+/*
+@QICore6Uplift: Update both Condition resources to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/KidneyHealthEvaluationFHIR.cql:100]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql [hunk 8]
```diff
@@ -113,6 +126,9 @@ define "Morbid Obesity Diagnosis or BMI Exam Result Greater Than or Equal to 40
   exists "Morbid Obesity Diagnosis On or Before Follow Up USS Assessment"
     or "BMI Exam Result Greater Than or Equal To 40 During Measurement Period and On or Before Follow Up USS Assessment"
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql:131]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql [hunk 9]
```diff
@@ -139,13 +155,19 @@ define "Has Initial BPH Diagnosis Starts During or Within 30 Days After End of H
       such that InitialBPHDiagnosis.prevalenceInterval ( ) starts during Interval[start of InHospitalServices.period, end of InHospitalServices.period + 31 days]
         and InHospitalServices.status = 'finished'
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql:160]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

File: input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql [hunk 10]
```diff
@@ -139,13 +155,19 @@ define "Has Initial BPH Diagnosis Starts During or Within 30 Days After End of H
...
 define "Initial BPH Diagnosis Starts Within 6 Months Before the Measurement Period":
   First([Condition: "Benign prostatic hyperplasia with lower urinary tract symptoms"] NewBPHDiagnosis
       where NewBPHDiagnosis.prevalenceInterval() starts during Interval[start of "Measurement Period" - 6 months, start of "Measurement Period"]
         and NewBPHDiagnosis.isConfirmedActiveDiagnosis()
       sort by start of onset.toInterval()
   )
 
+/*
+@QICore6Uplift: Update Condition to ConditionProblemsHealthConcerns
+*/
```
Comment (jbyrdevans) [inline input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql:170]: As written this should also be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis

**Review by jbyrdevans — CHANGES_REQUESTED** [review]

File: input/cql/Hospice.cql (hunk 2 above)
Comment (soy-park) [reply input/cql/Hospice.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql (hunk 4 above)
Comment (soy-park) [reply input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql (hunk 5 above)
Comment (soy-park) [reply input/cql/IntravesicalBacillusCalmetteGuerinForBladderCancerFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/KidneyHealthEvaluationFHIR.cql (hunk 6 above)
Comment (soy-park) [reply input/cql/KidneyHealthEvaluationFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/KidneyHealthEvaluationFHIR.cql (hunk 7 above)
Comment (soy-park) [reply input/cql/KidneyHealthEvaluationFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql (hunk 8 above)
Comment (soy-park) [reply input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql (hunk 9 above)
Comment (soy-park) [reply input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

File: input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql (hunk 10 above)
Comment (soy-park) [reply input/cql/UrinarySymptomScoreChangeAfterBenignProstaticHyperplasiaFHIR.cql]: Updated comment to be a union of ConditionProblemsHealthConcerns and ConditionEncounterDiagnosis.

**Pull request merged**